uqcsbot/__init__.py

```diff
@@ -60,25 +61,43 @@ def get_test_bot_token():
     '''
     Pings a channel on the UQCSTesting Slack that contains all the available
     bots, and Mitch. We can poll this channel to find  bots which are 'away'
-    (that is, not currently being used by anyone else) and return their
-    respective BOT_TOKEN.
+    (that is, not currently being used by anyone else)
+
+    Returns the bot user id and token in a dictionary: {'id': ..., 'token': ...}
     '''
     api_url = 'https://slack.com/api/conversations.members'
     response = requests.get(api_url, params={'token': API_TOKEN, 'channel': SECRET_BOT_MEETING_ROOM})
     if response.status_code != requests.codes['ok']:
-        print(f'Error: received status code {response.status.code}')
+        logger.error(f'Received status code {response.status.code}')
         sys.exit(1)
 
     json_contents = json.loads(response.content)
     if not json_contents['ok']:
-        print('Error: {0}'.format(json_contents['error']))
+        logger.error('{0}'.format(json_contents['error']))
         sys.exit(1)
 
     for user_id in json_contents['members']:
         if is_available_bot(user_id):
-            return BOT_TOKENS.get(user_id, None)
+            return {'id': user_id, 'token': BOT_TOKENS.get(user_id, None)}
     return None
 
+def get_test_bot_name(id):
+    '''
+    Get's a bot's name from their id
+    '''
+    api_url = 'https://slack.com/api/users.info'
+    response = requests.get(api_url, params={'token': API_TOKEN, 'user': id})
+    if response.status_code != requests.codes['ok']:
+        logger.error(f'Received status code {response.status.code}')
+        sys.exit(1)
+
+    json_contents = json.loads(response.content)
+    if not json_contents['ok']:
+        logger.error('{0}'.format(json_contents['error']))
+        sys.exit(1)
```

Comment:
we should probably replace the lazy `sys.exit(1)` with proper exception throwing and then just handle it below when we first call `get_test_bot()`

Reply:
No idea how to do exceptions in Python well. Is it overkill to write a custom exception type?